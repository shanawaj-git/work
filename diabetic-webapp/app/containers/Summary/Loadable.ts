/**
 *  Asynchornoulsy loads the component for Summary Screen
 */

import loadable from '../../utils/loadable';
export default loadable(() => import('./index'));
