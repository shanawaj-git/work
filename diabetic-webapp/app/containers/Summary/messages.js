import { defineMessages } from 'react-intl';
export default defineMessages({
  headingTitlePrefixLabel: {
    id: 'heading.title.prefix',
    defaultMessage: 'Prescription Sent',
  },
  headingTitleSuffixLabel: {
    id: 'heading.title.suffix',
    defaultMessage: 'for Approval',
  },
  descriptionLabel: {
    id: 'description.heading.label',
    defaultMessage: 'Your prescription has been sent to Allianz for approval',
  },
  boldDescriptionLabel: {
    id: 'boldDescription.heading.label',
    defaultMessage:
      'We notify you via text message on the status of your insurance. Post\n' +
      '            which you can place your order for delivery.',
  },
});
