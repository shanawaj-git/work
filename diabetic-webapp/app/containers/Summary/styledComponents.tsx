import styled from 'styled-components';
import { Typography } from '@albathanext/design-system';

export const Container = styled.div`
  background-color: #fafafa;
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100vh;
`;

export const HeaderDescription = styled.div`
  display: block;
  font-family: 'Neue Haas Grotesk Display Pro';
  font-style: normal;
  font-weight: 500;
  font-size: 32px;
`;
export const HeaderImageContainer = styled.div`
  direction: rtl;
`;

export const HeaderImage = styled.img`
  height: 460px;
  width: 315px;
`;

export const Description = styled.div`
  display: flex;
  flex-direction: column;
  margin-right: 30px;
  margin-left: 20px;
  .MuiTypography-root.MuiTypography-h1.suffix-title {
    font-size: 28px;
    margin-bottom: 18px;
  }
  .MuiTypography-root.MuiTypography-root.bold-text {
    color: #020a0a;
    font-weight: bold;
    font-style: normal;
    font-weight: 600;
    font-size: 15px;
    margin-top: 18px;
    line-height: 22px;
    margin-bottom: 2%;
  }
`;
