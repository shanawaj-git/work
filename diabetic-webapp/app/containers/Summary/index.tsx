import React from 'react';
import { any, InferProps } from 'prop-types';
import { injectIntl } from 'react-intl';
import { Typography, TypographyType } from '@albathanext/design-system';
import {
  Container,
  Description,
  HeaderDescription,
  HeaderImage,
  HeaderImageContainer,
} from './styledComponents';
import messages from './messages';
const education3 = require('images/educationImage3.svg');

function Summary({ intl }: SummaryScreenTypes) {
  return (
    <Container>
      <HeaderImageContainer>
        <HeaderImage src={education3} alt="header-img" />
      </HeaderImageContainer>
      <Description>
        <HeaderDescription>
          <Typography typographyType={TypographyType.TITLE_1}>
            {intl.formatMessage(messages.headingTitlePrefixLabel)}
          </Typography>
          <Typography
            typographyType={TypographyType.LARGE_TITLE}
            className="suffix-title"
          >
            {intl.formatMessage(messages.headingTitleSuffixLabel)}
          </Typography>
        </HeaderDescription>
        <Typography typographyType={TypographyType.SUBTITLE_1}>
          {intl.formatMessage(messages.descriptionLabel)}
        </Typography>
        <Typography
          typographyType={TypographyType.SUBTITLE_1}
          className="bold-text"
        >
          {intl.formatMessage(messages.boldDescriptionLabel)}
        </Typography>
      </Description>
    </Container>
  );
}
const SummaryScreenPropTypes = {
  intl: any,
};

type SummaryScreenTypes = InferProps<typeof SummaryScreenPropTypes>;

Summary.prototype = SummaryScreenPropTypes;

export default injectIntl(Summary);
