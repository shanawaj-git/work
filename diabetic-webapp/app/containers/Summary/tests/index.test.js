import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import React from 'react';
import Summary from '../index';

describe('<Summary/>', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <Summary />
      </IntlProvider>,
    );

    expect(firstChild).toMatchSnapshot();
  });
});
