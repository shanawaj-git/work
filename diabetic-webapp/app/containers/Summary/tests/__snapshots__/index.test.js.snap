// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary/> should render and match the snapshot 1`] = `
<div
  class="sc-bdVaJa iNQdlQ"
>
  <div
    class="sc-htpNat clYgmQ"
  >
    <img
      alt="header-img"
      class="sc-bxivhb cCoFGA"
      src="IMAGE_MOCK"
    />
  </div>
  <div
    class="sc-ifAKCX gbuHPh"
  >
    <div
      class="sc-bwzfXH hiyxzn"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-1jfg05l-MuiTypography-root"
      >
        Prescription Sent
      </h2>
      <h1
        class="MuiTypography-root MuiTypography-h1 suffix-title css-qu1isr-MuiTypography-root"
      >
        for Approval
      </h1>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1reifv3-MuiTypography-root"
    >
      Your prescription has been sent to Allianz for approval
    </p>
    <p
      class="MuiTypography-root MuiTypography-body1 bold-text css-1reifv3-MuiTypography-root"
    >
      We notify you via text message on the status of your insurance. Post
            which you can place your order for delivery.
    </p>
  </div>
</div>
`;
